Sources/ArgumentParser/Parsable Properties/Option.swift

```diff
@@ -498,24 +533,28 @@ extension Option {
 
 // MARK: - @Option Optional<T> Initializers
 extension Option {
-  /// Creates a property with a default value provided by standard Swift default
-  /// value syntax, parsing with the given closure.
+  /// Creates an optional property that reads its value from a labeled option,
+  /// parsing with the given closure, with an explicit `nil` default.
```

Comment:
In general, maybe yes. But it would also work with non-nil default, like:
```
var char: Character? = " "
```
The fact that it's a mutable optional would then allow it to eventually become nil later in time (when a program needs to free some memory for instance).

Reply:
That way of declaring (optional with a non-`nil` default) ends up using a different initializer that has a deprecation warning, since the behavior usually isn't what the author intends. I hadn't considered the use case you're describing here, but in general the ArgumentParser API is designed to essentially provide a way to write a function interface via the command line; the type declarations should match the CL interface.